### ISSUE TITLE
UX issues for users new to XMPP

### ISSUE THREAD

**Issue by trwnh** [siacs/Conversations#3440]
As an experiment, I just had various family members download and install Conversations from the Google Play Store, with their task being to send me a message, and giving them only their login details and the hostname. This has revealed the following:

1) Several of them were confused with how to proceed from the first screen. "Use my own provider" is confusing language when most non-technical users expect a "sign in" link instead. It's also non-obvious due to being greyed out.
2) The checkbox to "register an account on this server" is straight to the point, but one of my family members checked the box in confusion.
3) One family member was stuck on the "Start conversation" screen and unsure how to add my address as a contact. I had to manually intervene and guide them through adding a contact and starting a message, and this still took several attempts. They got stuck in "Manage accounts" several times, and also accidentally created a MUC with me. 

Suggestions:

1-2) Rework the welcome flow to not ask too much of the user at any one time. For example, this flow would probably work better:

```
Create account | Sign in 

If "Create account" is selected:
Use conversations.im | Use my own provider

    If "Use conversations.im" is selected:
    - Allow users to select a username and password

    If "Use my own provider" is selected:
    - Allow users to select a username, domain (curated list of open in-band-reg servers, or "custom"), and password

If "Sign in" is selected:
- Ask for username@domain and password
```

3) Use in-app tutorials to guide first-time users into adding their first contact. Maybe don't include 4 different actions in the same FAB.

**Comment by Echolon** [comment]
I can confirm all three cases. Happened to my contacts as well

**Comment by licaon-kter** [comment]
1. is a bit of a dark-pattern yes, given that the app is a way to get new paying users for the main server. Not sure how it can be improved yet still keep that.

2. Maybe rephrase as _"register a NEW account on this server"_ ?

3. There's a back arrow in accounts...what do you propose? I like the big single "ADD" button....spreading actions over more buttons would be more confusing, imho.

**Comment by iNPUTmice** [comment]
Splitting 'create new account' and 'use existing account' is actually a good idea.

As explained several times already we can’t provide a curated list of servers unless someone steps forward and creates and more importantly actively maintains one. People always want one; nobody wants to create one.

**Comment by trwnh** [comment]
@iNPUTmice What about using the list of 100% compliant servers from compliance.conversations.im?

**Comment by licaon-kter** [comment]
@trwnh That list doesn't show which ones allow open registration, so it's a lottery for a new user. Also, just because it's 100% doesn't mean anything else, eg. Policy, Terms etc.

**Comment by trwnh** [comment]
@licaon-kter Is it possible to test for in-band registration XEP?

**Comment by licaon-kter** [comment]
@trwnh ...maybe... https://github.com/iNPUTmice/caas/issues/13

**Comment by iNPUTmice** [comment]
> What about using the list of 100% compliant servers from compliance.conversations.im?

Compliance is only one of many criteria. Other criteria include but are not limited to

* Reliability ([status.conversations.im](https://status.conversations.im/historical/) can help)
* Do they have a restrictive cipher policy that excludes old Android phones
* Is there a privacy policy
* Is the administrator an incompetent asshole

That later two can probably not be generated automatically. That’s why the list needs to be _curated_ and not just _generated_.

**Comment by licaon-kter** [comment]
On the first screen, do remove the "For your convenience....conversations.im" text as it's already shown on the second one, and the note with the price is now missing
